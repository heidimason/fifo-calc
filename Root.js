import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { combineReducers } from 'redux'

import PurchaseHistoryReducer from './src/reducers/PurchaseHistory'
import SaleHistoryReducer from './src/reducers/SaleHistory'

const rootReducer = combineReducers({
		purchaseHistory: PurchaseHistoryReducer,
		saleHistory: SaleHistoryReducer
	}),

	store = createStore(
		rootReducer
	)

export default props => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
