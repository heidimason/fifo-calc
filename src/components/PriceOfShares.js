import React, { Component } from 'react'
import { Platform, TextInput, View } from 'react-native'
import { grayDark } from '../utils/styles/colors'
import { fonts } from '../utils/styles/fonts'
import { forms } from '../utils/styles/forms'

class PriceOfShares extends Component {
	handlePriceChange = price => {
		const { onPriceChange } = this.props

		onPriceChange(price)
	}

	render() {
		const { sharePrice } = this.props

		return (
			<TextInput
		        keyboardType="numeric"
				onChangeText={this.handlePriceChange}
				placeholder="Price per Share ($)"
				selectionColor={grayDark}
	        	style={[
	        		Platform.OS === 'ios'
	        		? forms.inputIOS
	        		: forms.inputAndroid,
					[fonts.h2, forms.textInput]
				]}
				value={sharePrice}
	        />
		)
	}
}

export default PriceOfShares