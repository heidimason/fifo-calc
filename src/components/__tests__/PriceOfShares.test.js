import React from 'react'
import renderer from 'react-test-renderer'

import Root from '../../../Root'
import PriceOfShares from '../PriceOfShares'

describe('<PriceOfShares />', () => {
	const tree = renderer.create(
		<Root>
			<PriceOfShares />
		</Root>
	).toJSON()

	it('renders correctly', () => {
		expect(tree).toMatchSnapshot()
	})
})
