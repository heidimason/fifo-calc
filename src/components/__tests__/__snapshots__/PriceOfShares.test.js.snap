// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceOfShares /> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="numeric"
  placeholder="Price per Share"
  rejectResponderTermination={true}
  selectionColor="#222"
  style={
    Array [
      Object {
        "marginBottom": 15,
      },
      Object {
        "borderRadius": 7,
      },
      Object {
        "borderColor": "transparent",
        "borderWidth": 0,
      },
      Array [
        Object {
          "fontSize": 40,
        },
        Object {
          "backgroundColor": "#fff",
          "color": "#aaa",
          "height": 45,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ],
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
